File: src/transformers/features/json/JsonValidateParseTransformer.ts
import { JsonValidateParseProgrammer } from "../../../programmers/json/JsonValidateParseProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonValidateParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "json.validateParse",
      write: JsonValidateParseProgrammer.write,
    });
}

File: packages/typia/native/transform/features/json/JsonValidateParseTransformer.go
package json

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativejsonprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/json"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type jsonValidateParseTransformerNamespace struct{}

var JsonValidateParseTransformer = jsonValidateParseTransformerNamespace{}

func (jsonValidateParseTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "json.validateParse",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativejsonprogrammers.JsonValidateParseProgrammer.Write(x)
		},
	})
}
